src/main/java/sirius/biz/process/ProcessEnvironment.java

```diff
@@ -194,9 +198,27 @@ public void updateTitle(String newTitle) {
 
     @Override
     public void log(ProcessLog logEntry) {
+        if (Strings.isFilled(logEntry.getMessageType()) && logEntry.getMaxMessagesToLog() > 0) {
+            AtomicInteger messagesSoFar =
+                    messageCountsPerType.computeIfAbsent(logEntry.getMessageType(), this::countMessagesForType);
+            messagesSoFar.incrementAndGet();
+            if (messagesSoFar.incrementAndGet() >= logEntry.getMaxMessagesToLog()) {
+                return;
+            }
+            if (messagesSoFar.incrementAndGet() > logEntry.getMaxMessagesToLog()) {
+                return;
+            }
```

Comment:
Does this code block make sense? Shouldn't it just be the bottom if-block?

Reply:
+1